ament_index_cpp/src/get_resources.cpp

```
@@ -46,10 +46,18 @@ get_resources(const std::string & resource_type)
     }
     dirent * entry;
     while ((entry = readdir(dir)) != NULL) {
-      // ignore folders starting with a dot
+      // ignore directories/files starting with a dot
       if (entry->d_name[0] == '.') {
         continue;
       }
+
+      // ignore directories
+      auto subdir = opendir((path + "/" + entry->d_name).c_str());
```

Should all the error cases just be silently ignored or should the code warn about some of them?